ffs/interfaces.go

```diff
@@ -54,9 +54,11 @@ type Scheduler interface {
 // data or Cids.
 type HotStorage interface {
 	Add(context.Context, io.Reader) (cid.Cid, error)
+	Remove(context.Context, cid.Cid) error
 	Get(context.Context, cid.Cid) (io.Reader, error)
 	Pin(context.Context, cid.Cid) (int, error)
 	Put(context.Context, blocks.Block) error
+	IsStored(context.Context, cid.Cid) (bool, error)
```

Comment:
Nit: `Pin` related to `IsStored`, so maybe `IsStored` -> `Pinned` or `IsPinned`, or since the API is called `HotStorage`, `Pin` -> `Store`?

Reply:
Yeah, I didn't want to use a `Pin` concept because `HotStorage` should be a higher-level concept than IPFS.
Say, a `HotStorage` could be saving things in S3 (or in both IPFS and S3, who knows).

Reply:
If I remember right, `Pin` is the method to store data that is already in IFPS, take the existing cid and _pin_ it here. But you can also `Add` (and maybe `Put`, I'm not sure), raw data. So there are two ways (at least) of storing data. `IsStored` is how you check if data stored in any of those ways is stored.

Reply:
> Yeah, I didn't want to use a `Pin` concept because `HotStorage` should be a higher-level concept than IPFS.
> Say, a `HotStorage` could be saving things in S3 (or in both IPFS and S3, who knows).

Agreed. That's why I was suggesting the option of not using the word "pin" in the API.

Reply:
Taking the buckets case, the only reason to not auto-pin when you add is because it's more efficient to _update_ the existing pin (if it exists). Ideally, that would be handled internally in add/put:
1. Search for a pin for the given cid/block, if exists, add w/o auto-pinning
2. Add/put
3. If there was an existing pin, update it

Reply:
(Definitely not blocking this PR, just thinking out loud)

Reply:
Sorry @sanderpick , I misread your original comment; now I see what you mean.
First, I agree that `Pin()` should be changed to `Store` to be aligned with `IsStored`. I'll add this change to the next PR.

Regarding `Put()`, the only use of that API is for CAR importing. The CAR library needs a `Store` to store blocks when the CAR file is processed:
```go
type Store interface {
	Put(blocks.Block) error
}
```
After all that is loaded, then `Pin(..)` is called with the root of the DAG, which pins recursively.

I preferred not pinning in that implementation because if the CAR importing fails in the middle of the process, that would mean that half of the DAG was pinned without its root. And pinning blocks without the proper root would mean that will be pinned forever, since unpinning is with the root.

To be clear, that API is called on individual blocks of the DAG by the CAR lib, not only with root Blocks. That's why this is different than _buckets_. 
Also, at the FFS level the concept of _Updating_ a pin doesn't _make sense_ since the API pins Cids, not _Files_ or higher-level concepts that mutate. 

For some time I've been thinking about the need of a new API to `Swap` or `Replace` or a similar idea to simplify some client use-cases as _Buckets_ or similar. They can still add and remove Cids, but native support for `Swap`ping Cids makes the Pin update at the IPFS layer more efficient.

Makes sense?

Reply:
Okay, cool! Makes sense.